### ISSUE TITLE
Agent Pool with multiple agents using always same agent

### ISSUE THREAD

**Issue by ghost** [microsoft/azure-pipelines-agent#3054]
## Agent Version and Platform
Version of your agent? 2.144.0/2.144.1/...
2.172.1
OS of the machine running the agent? OSX/Windows/Linux/...
Windows 10 x64

## Azure DevOps Type and Version
dev.azure.com
https://dev.azure.com/AngelServera0506/

## What's not working?
I have an agent pool with four agents running on one VM (Windows 10) and one laptop(Windows 10) for testing purposes.
I have run four pipelines with one or two jobs using the same Agent Pool, but the jobs are always running sequentially on the first registered agent in the pool. In other words, the jobs are queued until the agent 1 finishes with the previous job, and in the meantime, the rest of the agents are listening but doing nothing. is that normal or I am missing something?
If I stooped the agent 1 the second in the order to connect, attends the jobs and run jobs fine, so it seems that the agent pool is working on Failover strategy with one active and rest on passive mode, but I would like to work on active all, so all the agents are attending jobs based on their workload.

**Comment by aservera** [comment]
​Hi,
I have made more tests and identify the origin of the strange behavior. The initial test was launching a simple Maven pipeline under classic mode. However, yesterday I did more tests using a yaml one, with multiple jobs and forcing the parallelism and it worked as expected, balancing the jobs between all available nodes.
So from my PoV the issue is only with classic one, probably also with Release pipelines.
Best
